### PR TITLE
chore: update wasmtime to 6.0, bump extism versions

### PR DESCRIPTION
**Pull request by nilslice**
Unsure if now is the best time to do the `extism` crate version bumps, but I figured they'd need to happen at some point in the near future anyways. Happy to revert if there is any opposition. Also, I added back the local `path` property to the Elixir NIF crate manifest. I want to see if this breaks again in CI, or if we can leave it.

### CODE REVIEW

**Comment by zshipko** [comment]
Nice! The issue with the extism_nif crate is in the release workflow. I'm not sure what the best way to toggle that path argument is, but for now it's helpful to make sure the new changes are working with the elixir sdk.

**Comment by zshipko** [comment]
@steve: just updated this to work with wasmtime 6.0.0, I think we could merge this (if you think the other changes on this branch are good to go)

**Comment by zshipko** [comment]
@nilslice: just updated this to work with wasmtime 6.0.0, i think we can merge this if you think the other changes on this branch are good to go

**Comment by zshipko** [comment]
the zig failure looks like its related to some language changes, i can update that in another PR

**Comment by nilslice** [comment]
> the zig failure looks like its related to some language changes, i can update that in another PR

@zshipko, yea I think you are right! sounds good to me :shipit:

**Comment by nilslice** [comment]
I can't approve it though since I opened it initially. Please go ahead and do so at your convenience :)

**Review by zshipko — APPROVED** [review]

**Pull request merged**
